services/provider/server/server.go

```diff
@@ -6,65 +6,185 @@ import (
+		}
+		return nil, status.Errorf(codes.Internal, "failed to create storageConsumer %q. %v", req.ConsumerName, err)
+	}
+
+	// TODO: send correct granted capacity
+	return &pb.OnboardConsumerResponse{StorageConsumerUUID: storageConsumerUUID, GrantedCapacity: req.Capacity}, nil
 }
 
 // GetStorageConfig RPC call to onboard a new OCS consumer cluster.
-func (c *ocsProviderServer) GetStorageConfig(ctx context.Context, req *pb.StorageConfigRequest) (*pb.StorageConfigResponse, error) {
+func (s *OCSProviderServer) GetStorageConfig(ctx context.Context, req *pb.StorageConfigRequest) (*pb.StorageConfigResponse, error) {
 	mock := os.Getenv(common.MockProviderAPI)
 	if mock != "" {
 		return mockGetStorageConfig(common.MockError(mock))
 	}
-	return &pb.StorageConfigResponse{}, nil
-}
 
-// OffboardConsumer RPC call to delete the StorageConsumer CR
-func (c *ocsProviderServer) OffboardConsumer(ctx context.Context, req *pb.OffboardConsumerRequest) (*pb.OffboardConsumerResponse, error) {
-	mock := os.Getenv(common.MockProviderAPI)
-	if mock != "" {
-		return mockOffboardConsumer(common.MockError(mock))
+	// Get storage consumer resource using UUID
+	consumerObj, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, err
 	}
-	return &pb.OffboardConsumerResponse{}, nil
+
+	// Verify Status
+	switch consumerObj.Status.State {
+	case ocsv1alpha1.StorageConsumerStateFailed:
+		// TODO: get correct error message from the storageConsumer status
+		return nil, status.Errorf(codes.Internal, "storageConsumer status failed")
+	case ocsv1alpha1.StorageConsumerStateConfiguring:
+		return nil, status.Errorf(codes.Unavailable, "waiting for the rook resources to be provisioned")
+	case ocsv1alpha1.StorageConsumerStateDeleting:
+		return nil, status.Errorf(codes.NotFound, "storageConsumer is already in deleting phase")
+	case ocsv1alpha1.StorageConsumerStateReady:
+		// TODO: Return json connection details
+		return &pb.StorageConfigResponse{}, nil
+	}
+
+	return nil, status.Errorf(codes.Unavailable, "storage consumer status is not set")
 }
 
 // UpdateCapacity PRC call to increase or decrease the storage pool size
-func (c *ocsProviderServer) UpdateCapacity(ctx context.Context, req *pb.UpdateCapacityRequest) (*pb.UpdateCapacityResponse, error) {
+func (s *OCSProviderServer) UpdateCapacity(ctx context.Context, req *pb.UpdateCapacityRequest) (*pb.UpdateCapacityResponse, error) {
 	mock := os.Getenv(common.MockProviderAPI)
 	if mock != "" {
 		return mockUpdateCapacity(common.MockError(mock))
 	}
+
+	capacity, err := resource.ParseQuantity(req.Capacity)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "%q is not a valid resource capacity: %v", req.Capacity, err)
+	}
+
+	err = s.consumerManager.UpdateCapacity(ctx, req.StorageConsumerUUID, capacity)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to update capacity in the storageConsumer resource: %v", err)
+	}
+
```

Comment:
I think it should also take care of if CR not found and set appropriate code accordingly

Reply:
sending `NotFound` status code if storageConsumer is not found while updating capacity